docs/doc/60-contributing/03-rfcs/20221013-iceberg-catalog.md

```diff
@@ -0,0 +1,236 @@
+| `double`        | 64 bit IEEE 754 float point number                       | `DOUBLE`                             |
+| `decimal(P, S)` | Fixed point decimal; precision P, scale S                | not supported                        |
+| `date`          | Calendar date without timezone or time                   | `DATE`                               |
+| `time`          | Timestamp without date, timezone                         | `TIMESTAMP`, convert date to today   |
+| `timestamp`     | Timestamp without timezone                               | `TIMESTAMP`, convert timezone to GMT |
+| `timestamptz`   | Timestamp with timezone                                  | `TIMESTAMP`                          |
+| `string`        | UTF-8 string                                             | `VARCHAR`                            |
+| `uuid`          | 16-byte fixed byte array, universally unique identifiers | `VARCHAR`                            |
+| `fixed(L)`      | Fixed-length byte array of length L                      | `VARCHAR`                            |
+| `binary`        | Arbitrary-length byte array                              | `VARCHAR`                            |
+| `struct`        | a tuple of typed values                                  | `OBJECT`                             |
+| `list`          | a collection of values with some element type            | `ARRAY`                              |
+| `map`           |                                                          | `OBJECT`                             |
+
+## Drawbacks
+
+None
+
+## Rationale and alternatives
+
+### Iceberg External Table
+
+Creating an external table from Iceberg storage:
+
+```sql
+CREATE EXTERNAL TABLE [IF NOT EXISTS] [db.]table_name
+[(
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    ...
+)] ENGINE=ICEBERG
+ENGINE_OPTIONS=(
+  DATABASE='db0'
+  TABLE='tbl0'
+  LOCATION=<external-location>
+)
+```
+
+A new table engine `ICEBERG` will be introduced, and all data will remain still in the Iceberg storage. The external table also should support users to query its snapshot data and time travelling.
+
+```sql
+SELECT snapshot_id FROM ICEBERG_SNAPSHOT('<db_name>', '<external_table_name');
+```
+
+```
+ snapshot_id
+--------------
+73556087355608
+```
+
+After discussion, the catalog way above is chosen, for its more complete support to Iceberg features, and it is more aligned to the current design of Hive catalog.
+
+### Create Table from Iceberg Snapshots
+
+Iceberg provides a snapshot capability and tables can be created from them.
+
+```sql
+CREATE TABLE [IF NOT EXISTS] [db.]table_name
+(
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    ...
+) ICEBERG_SNAPSHOT_LOCATION=
```

Comment:
I prefer to move those engine-related configs into `engine_options`.

Reply:
Creating tables from Iceberg Snapshots is an alternative way and will not be implemented. If this feature is good to pick then I'll move it to `Future Possibilities.

Reply:
Please do this.